regression_tests/test_full_state_checkpoint_signal.sh

```
@@ -0,0 +1,185 @@
+    kill -9 "$TRAIN_PID" 2>/dev/null
+    exit 1
+fi
+
+# Write trigger file (this is what signal handlers do in production)
+echo "=== Writing trigger file: $TRIGGER_FILE ==="
+touch "$TRIGGER_FILE"
+
+# Wait for clean exit
+echo "Waiting for training to checkpoint and exit..."
+EXITED=false
+for i in $(seq 1 120); do
```

Shellcheck (0.11.0)

[warning] i appears unused. Verify use (or export if used externally).

(SC2034)